pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java

```diff
@@ -35,4 +35,25 @@
    * @return Whether the client has data access to the table
    */
   boolean hasDataAccess(HttpHeaders httpHeaders, String tableName);
+
+  /**
+   * Return whether the client has write permission to the given table
+   *
+   * @param httpHeaders http headers
+   * @param tableName name of the table to be accessed
+   * @return whether the client has write permission
+   */
+  default boolean hasWritePermission(HttpHeaders httpHeaders, String tableName) {
```

Comment:
can we do hasAccess(AccessType, HttpHeaders httpHeaders, String tableName) this will be more flexible and we can add more types later without having to change the interface

Reply:
That's a good idea. I'll refactor.